common/frame_session.go

```diff
@@ -827,17 +845,19 @@ func (fs *FrameSession) onAttachedToTarget(event *target.EventAttachedToTarget)
 	// Handle new worker
 	w, err := NewWorker(fs.ctx, session, targetID, event.TargetInfo.URL)
 	if err != nil {
-		if !fs.page.browserCtx.browser.connected && strings.Contains(err.Error(), "websocket: close 1006 (abnormal closure)") {
-			// If we're no longer connected to browser, then ignore WebSocket errors
+		// ignore errors if we're no longer connected to browser
+		// this happens when there is no browser but we still want to
+		// attach a worker to it.
+		if !fs.page.browserCtx.browser.connected {
 			return
 		}
 		select {
 		case <-fs.ctx.Done():
 			fs.logger.Debugf("FrameSession:onAttachedToTarget:NewWorker:<-ctx.Done",
 				"sid:%v tid:%v esid:%v etid:%v ebctxid:%v type:%q",
 				fs.session.id, fs.targetID, event.SessionID,
-				event.TargetInfo.TargetID, event.TargetInfo.BrowserContextID, event.TargetInfo.Type)
-
+				event.TargetInfo.TargetID, event.TargetInfo.BrowserContextID,
+				event.TargetInfo.Type)
 			return
 		default:
 			k6Throw(fs.ctx, "cannot create frame session (worker): %w", err)
```

Comment:
Maybe unrelated to this PR, but the below `k6Throw()` (line 865) will never be reached, so we can remove it. (GitHub doesn't allow me to comment on that line...)

Reply:
Good catch!